db/python/layers/project_insights.py

```diff
@@ -301,367 +332,601 @@ async def _stats_es_indexes_query(
 WHERE a.project IN :projects
 AND JSON_UNQUOTE(JSON_EXTRACT(a.meta, '$.sequencing_type')) in :sequencing_types;
         """
-        latest_es_indexes_by_project_id_and_seq_type_and_stage = await self.connection.fetch_all(
+        _query_results = await self.connection.fetch_all(
             _query,
             {
-                'projects': projects,
-                'sequencing_types': sequencing_types,
+                "projects": project_ids,
+                "sequencing_types": sequencing_types,
             },
         )
-        return latest_es_indexes_by_project_id_and_seq_type_and_stage
-
-    async def _details_sequencing_groups_report_links(
-        self, projects: list[int], sequencing_types: list[str]
-    ):
-        """Get sequencing group web report links"""
+        latest_es_indices_by_project_id_and_seq_type_and_stage: defaultdict[
+            tuple[ProjectId, str, str], dict[str, Any]
+        ] = defaultdict(dict)
+        for row in _query_results:
+            latest_es_indices_by_project_id_and_seq_type_and_stage[
+                (row["project"], row["sequencing_type"], row["stage"])
+            ] = {
+                "id": row["id"],
```

Comment:
Maybe create a namedtuple, or tiny dataclass for this, be your own friend for typing